examples/ping-pong/src/main.rs

```diff
@@ -43,37 +41,34 @@ async fn main() -> ! {
     /// the handler itself is being passed into the Graphcast Agent, so it needs to be static as well.
     pub static GRAPHCAST_AGENT: OnceCell<GraphcastAgent> = OnceCell::new();
 
-    // subtopics are optionally provided and used as the content topic identifier of the message subject,
-    // if not provided then they are usually generated based on indexer allocations
-    let mut subtopics = vec!["ping-pong-content-topic".to_string()];
-    for topic in config.topics.clone() {
-        if !subtopics.contains(&topic) {
-            subtopics.push(topic);
-        }
-    }
+    let config = Config::args();
 
-    debug!("Initializing the Graphcast Agent");
-    let graphcast_agent = GraphcastAgent::new(
-        // wallet key can be either private key or mnemonic, resolves into ethereum wallet and indexer identity.
-        // Unwrap is okay here thanks to configuration validate_set_up
-        config.wallet_input().unwrap().to_string(),
-        // radio_name is used as part of the content topic for the radio application
+    let mut graphcast_agent_config = GraphcastAgentConfig::new(
+        config.private_key.expect("No private key provided"),
         radio_name,
-        &config.registry_subgraph,
-        &config.network_subgraph,
-        &config.graph_node_endpoint,
-        config.boot_node_addresses,
-        Some(&config.graphcast_network),
-        subtopics,
-        // Waku node address is set up by optionally providing a host and port, and an advertised address to be connected among the waku peers
-        // Advertised address can be any multiaddress that is self-describing and support addresses for any network protocol (tcp, udp, ip; tcp6, udp6, ip6 for IPv6)
-        config.waku_node_key,
-        config.waku_host,
-        config.waku_port,
+        config.registry_subgraph,
+        config.network_subgraph,
+        config.graph_node_endpoint.clone(),
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
         None,
     )
-    .await
-    .expect("Could not create Graphcast agent");
+    .unwrap_or_else(|e| panic!("Could not create GraphcastAgentConfig: {e}"));
+
+    graphcast_agent_config.graphcast_namespace = Some("testnet".to_string());
```

Comment:
these should be passed in directly at `GraphcastAgentConfig::new` and then `graphcast_agent_config` doesn't need to be `mut`